weldx/asdf/util.py

```diff
@@ -141,16 +142,16 @@ def write_read_buffer(
     return read_buffer(buffer, open_kwargs)
 
 
-def get_yaml_header(file, parse=False) -> Union[str, dict]:
+def get_yaml_header(file: Union[str, Path, BytesIO], parse=False) -> Union[str, dict]:
     """Read the YAML header part (excluding binary sections) of an ASDF file.
 
     Parameters
     ----------
-    file :
-        filename, ``pathlib.Path`` or ``BytesIO`` buffer of ASDF file
+    file : str, pathlib.Path or io.BytesIO
```

Comment:
I thought we agreed to infer parameters types from the type annotation?

Reply:
We could also think of a type alias for this thing, as lots of our functions dealing with file or path like types share this common thing.

Reply:
yes I just could not get `BytesIO` to work here (god knows why, it works in the other places) so typed it manually again

Reply:
BytesIO is in the namespace, right? I assume there should be no difference in sphinx then, when the types are just named/accessed in the same fashion.

Reply:
it is, somehow the intersphinx links mess it up and want to link `_io.BytesIO` (the C module)

feel free to give it a try locally I would be glad to have a solution (this popped up allready)